### DISCUSSION TITLE
Different ports for client and server connections

### DISCUSSION THREAD

**Discussion by ropucyka** [bluenviron/mediamtx#2694] [Questions]
## Describe the feature

In example with RTMP:

We listening for incoming (push from OBS etc.) connections on **1935** port, but we awaiting for clients connections (from Internet, LAN) also on same port. - wtf?
And mediamtx configured as **overridePublisher: _yes_** by default. Oh my god.

**Comment by aler9** [comment]
Hello, the situation is the same in 99% of media servers. In order to solve your issue, you can:

* prevent unauthorized publishers from publishing to the server by turning on authentication (`publishUser` / `publishPass`)
* use two MediaMTX instances, one for publishing, that listens on a publisher-specific port of your choice, and the other for reading, that takes its streams from the first instance. Configuration of the first instance:

   ```yml
   rtmpAddress: :1936
   ```

   Configuration of the second instance:

   ```yml
   paths:
     mypath:
       source: rtsp://first-instance/mypath
   ```

[a comment by an automated account is not shown]
